src/Common/ProxyConfiguration.h

```diff
@@ -30,7 +30,10 @@ struct ProxyConfiguration
             return Protocol::HTTPS;
         }
 
-        throw Exception(ErrorCodes::BAD_ARGUMENTS, "Unknown proxy protocol: {}", str);
+        if (str.empty())
+            throw Exception(ErrorCodes::BAD_ARGUMENTS, "Empty protocol in the URL");
+        else
+            throw Exception(ErrorCodes::BAD_ARGUMENTS, "Unknown protocol in the URL: {}", str);
```

Comment:
```suggestion

        throw Exception(ErrorCodes::BAD_ARGUMENTS, "Unknown protocol in the URL: {}", str);
```

Reply:
Ok, but does it matter? Let me wait for the CI, if it will be green, I will merge without this change, if not, I will commit this change :)